tests/test_allow_unmocked.js

```diff
@@ -2,239 +2,238 @@
 
 const http = require('http')
 const { expect } = require('chai')
-const { test } = require('tap')
 const nock = require('..')
 const got = require('./got_client')
 
-require('./cleanup_after_each')()
 require('./setup')
 
-test('with allowUnmocked, mocked request still works', async () => {
-  const scope = nock('http://example.test', { allowUnmocked: true })
-    .post('/')
-    .reply(200, '99problems')
+describe('allowUnmocked option', () => {
+  let _server
 
-  const { body, statusCode } = await got.post('http://example.test/')
-  expect(statusCode).to.equal(200)
-  expect(body).to.equal('99problems')
-
-  scope.done()
-})
-
-test('allow unmocked works after one interceptor is removed', async t => {
-  const server = http.createServer((request, response) => {
-    response.write('live')
-    response.end()
+  afterEach(() => {
+    if (_server) {
+      _server.close()
+      _server = null
+    }
   })
-  t.once('end', () => server.close())
-
-  await new Promise(resolve => server.listen(resolve))
-
-  const url = `http://localhost:${server.address().port}`
-
-  nock(url, { allowUnmocked: true })
-    .get('/')
-    .reply(200, 'Mocked')
 
-  expect((await got(url)).body).to.equal('Mocked')
-  expect((await got(url)).body).to.equal('live')
-})
+  const createServer = async requestListener => {
```

Comment:
Think we're mostly using this style.

I've attempted to lint on this in other projects and haven't found the way, though we could give that another look.

```suggestion
  async function createServer() {
```